vmicore/src/lib/vmi/InterruptEventSupervisor.cpp

```diff
@@ -58,10 +58,12 @@ namespace VmiCore
 
     std::shared_ptr<IBreakpoint>
     InterruptEventSupervisor::createBreakpoint(uint64_t targetVA,
-                                               uint64_t processDtb,
+                                               std::shared_ptr<const ActiveProcessInformation> processInformation,
                                                const std::function<BpResponse(IInterruptEvent&)>& callbackFunction,
                                                bool global)
     {
+        auto processDtb = targetVA >= PagingDefinitions::kernelspaceLowerBoundary ? processInformation->processKernelDTB
+                                                                                  : processInformation->processUserDTB;
```

Comment:
`processUserDTB` will be 0 if KPTI is disabled. I would suggest to check for that in `extractProcessInformation` and set `processUserDTB` to the same value as `processKernelDTB` if that is the case.

Reply:
userDirectoryTableBase will now be initialized with the same offset as directoryTableBase if the version is not implementing KPTI.In that case processUserDTB should contain the processKernelDTB, if im not mistaken.

Reply:
You misunderstand. Even if the system has KPTI implemented, it may still decide not to use it (you can verify this behavior on our kvm test machine) if it determines that the CPU is not vulnerable to Meltdown. In this case, the user DTB is present but its content is 0.